tests/core/executor/test_SelfTargetExecutor.py

```diff
@@ -27,9 +38,161 @@ def test_self_target_executor_init_raises_error_when_target_type_is_not_self(sel
+                    virtualenv_command="python3 -m venv",
+                ),
+                failed_when=["result_pip.state == 'absent'"],
+            ),
+            dict(
+                name="Install ychaos[agents]",
+                register="result_pip_install_ychaos_agents",
+                action=dict(
+                    module="pip",
+                    chdir="{{result_pwd.stdout}}",
+                    name="ychaos[agents]",
+                    virtualenv="ychaos_env",
+                ),
+                failed_when=["result_pip_install_ychaos_agents.state == 'absent'"],
+            ),
+            dict(
+                name="Create a workspace directory for storing local report files",
+                register="result_create_workspace",
+                action=dict(
+                    module="file",
+                    path="{{result_pwd.stdout}}/ychaos_ws",
+                    state="directory",
+                    mode="0755",
+                ),
+            ),
+            dict(
+                name="Put testplan in ychaos_ws",
+                register="result_testplan_file",
+                action=dict(
+                    module="copy",
+                    content=json.dumps(
+                        mock_valid_testplan.to_serialized_dict(), indent=4
+                    ),
+                    dest="{{result_create_workspace.path}}/testplan.json",
+                ),
+            ),
+            dict(
+                name="Run YChaos Agent",
+                ignore_errors="yes",
+                action=dict(
+                    module="shell",
+                    args=dict(
+                        cmd=" ".join(
+                            [
+                                "ychaos --log-file '{{result_create_workspace.path}}/ychaos.log'",
+                                "agent attack --testplan '{{result_testplan_file.dest}}' --attack-report-yaml '{{result_create_workspace.path}}/attack_report.yaml'",
+                            ]
+                        ),
+                    ),
+                ),
+            ),
+            dict(
+                name="Delete Virtual environment",
+                action=dict(
+                    module="file",
+                    path="{{result_pip.virtualenv}}",
+                    state="absent",
+                ),
+            ),
+        ]
+        playbook_tasks = executor.ansible_context.play_source["tasks"]
+
+        self.assertEqual(len(playbook_tasks), len(expected_tasks))
+        for i, task in enumerate(playbook_tasks):
+            self.assertDictEqual(task, expected_tasks[i])
+
+    def test_self_executor_execute_with_no_errors_running_playbook(self):
+        mock_valid_testplan = TestPlan.load_file(
+            self.testplans_directory.joinpath("valid/testplan1.yaml")
+        )
+        executor = SelfTargetExecutor(mock_valid_testplan)
+        executor.prepare()
+
+        when(executor.ansible_context.tqm).run(ANY).thenReturn(True)
```

Comment:
```suggestion
        expect(executor.ansible_context.tqm, times=1).run(ANY).thenReturn(True)
```
https://mockito-python.readthedocs.io/en/latest/the-functions.html#mockito.expect